homeassistant/components/shelly/__init__.py

```diff
@@ -24,25 +23,34 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-async def async_setup(hass: HomeAssistant, config: dict):
-    """Set up the Shelly component."""
-    hass.data[DOMAIN] = {DATA_CONFIG_ENTRY: {}}
-    hass.data[DOMAIN][COAP_CONTEXT] = await aiocoap.Context.create_client_context()
+async def get_coap_context(hass):
+    """Init COAP context if not exists."""
+    if DOMAIN not in hass.data:
+        hass.data.setdefault(DOMAIN, {})
+    if COAP_CONTEXT not in hass.data[DOMAIN]:
+        hass.data[DOMAIN][COAP_CONTEXT] = aioshelly.COAP()
+        await hass.data[DOMAIN][COAP_CONTEXT].initialize()
 
     async def shutdown_listener(*_):
         """Home Assistant shutdown listener."""
-        await hass.data[DOMAIN][COAP_CONTEXT].shutdown()
+        hass.data[DOMAIN][COAP_CONTEXT].close()
 
     hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, shutdown_listener)
 
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Shelly component."""
+    await get_coap_context(hass)
+    hass.data[DOMAIN][DATA_CONFIG_ENTRY] = {}
```

Comment:
This is weird. `hass.data[DOMAIN` doesn't exist.

Reply:
This was indirectly created by the "get_coap_context", fixed